### PR TITLE
feat(content-item-horizontal): update spacing and add media option to default

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#9436 

### Description

Multiple updates to content item horizontal variations:

Screen Shot 2022-10-10 at 11 54 52 AM

### Changelog

**New**

- media option to default content-item-horizontal

**Changed**

- removed unnessary knobs from stories
- updated spacing to align with specs
- use the 12 column grid instead of 10

### CODE REVIEW

**Comment by oliviaflory** [comment]
@ariellalgilmore This looks great! 

The only thing I noticed is in the Thumbnail variant, the image should span 4 columns at the medium breakpoint. Maybe barely a thumbnail at that point...but we will try it out!

Screen Shot 2022-10-17 at 11 40 43 AM
Screen Shot 2022-10-17 at 3 47 52 PM

**Review by RichKummer — APPROVED** [review]

File: packages/styles/scss/components/content-item-horizontal-media/_content-item-horizontal-media.scss [hunk 1]
```diff
@@ -156,10 +155,10 @@
 
     .#{$prefix}--content-item-horizontal__row {
       + .#{$prefix}--content-item-horizontal__row {
-        margin-block-start: $carbon--spacing-05;
+        // margin-block-start: $carbon--spacing-05;
```
Comment (emyarod) [inline packages/styles/scss/components/content-item-horizontal-media/_content-item-horizontal-media.scss:158]: do we want to keep the comments?

File: packages/styles/scss/components/content-item-horizontal-media/_content-item-horizontal-media.scss [hunk 2]
```diff
@@ -156,10 +155,10 @@
...
 
         @include carbon--breakpoint(md) {
-          margin-block-start: $carbon--spacing-07;
+          // margin-block-start: $carbon--spacing-07;
```
Comment (emyarod) [inline packages/styles/scss/components/content-item-horizontal-media/_content-item-horizontal-media.scss:161]: same as above

**Review by emyarod — APPROVED** [review]
looks good to me! just one question

**Pull request merged**